src/com/amazon/ion/IonBufferConfiguration.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.ion;
+
+/**
+ * Configures Ion lookahead buffers.
+ */
+public final class IonBufferConfiguration extends BufferConfiguration<IonBufferEventHandler, IonBufferConfiguration> {
+
+    /**
+     * Builds IonBufferConfiguration instances.
+     */
+    public static final class Builder
+        extends BufferConfiguration.Builder<IonBufferEventHandler, IonBufferConfiguration, Builder> {
+
+        /**
+         * 4-byte IVM + 1 byte user value.
+         */
+        private static final int MINIMUM_MAX_VALUE_SIZE = 5;
+
+        /**
+         * An IonBufferEventHandler that does nothing.
+         */
+        private static final IonBufferEventHandler NO_OP_EVENT_HANDLER = new IonBufferEventHandler() {
+
+            @Override
+            public void onOversizedSymbolTable() {
+                // Do nothing.
+            }
+
+            @Override
+            public void onOversizedValue() {
+                // Do nothing.
```

Comment:
Silently moving on is fine if the user asks for/expects that behavior, but I think the default for this should be an exception. That would surface this as a problem to anyone who hasn't carefully read the docs.

Reply:
Similar reasoning to the question about `onOversizedSymbolTable`. The user has to provide a handler if it's possible for an oversized value to be encountered.

Reply:
Same thought here: makes total sense, tweaking the comment would help allay potential confusion.

Reply:
Done.